core/src/components/button/button.ionic.scss

```diff
@@ -53,45 +53,61 @@
   --ripple-color: var(--background-activated);
 }
 
+:host(.button-solid.ion-color) {
+  --background-activated: #{globals.current-color(shade)};
+  --background-hover: #{globals.current-color(shade)};
+  --background: #{globals.current-color(base)};
+  --color: #{globals.current-color(contrast)};
+}
+
 // Outline Button
 // --------------------------------------------------
 
 :host(.button-outline) {
   --border-width: #{globals.$ion-border-size-025};
   --border-style: #{globals.$ion-border-style-solid};
-  --background-activated: #{globals.$ion-bg-neutral-subtlest-press};
+  --background-activated: #{globals.$ion-bg-primary-subtle-press};
   --background-focused: transparent;
-  --background-hover: #{globals.$ion-bg-neutral-subtlest-press};
+  --background-hover: #{globals.$ion-bg-primary-subtle-press};
   --background-hover-opacity: 1;
   --border-color: #{globals.ion-color(primary, base)};
   --color: #{globals.ion-color(primary, base)};
   --ripple-opacity: var(--background-activated-opacity, 1);
   --ripple-color: var(--background-activated);
 }
 
+:host(.button-outline.ion-color) {
+  --background-activated: #{globals.current-color(shade, $subtle: true)};
+  --background-hover: #{globals.current-color(shade, $subtle: true)};
+  --border-color: #{globals.current-color(base)};
+  --color: #{globals.current-color(base)};
+}
+
+// Warning Outline Button - use foreground color for text and border
+:host(.button-outline.ion-color-warning) .button-native {
+  --ion-color-base: #{globals.ion-color(warning, foreground)};
+}
+
 // Clear Button
 // --------------------------------------------------
 
 :host(.button-clear) {
-  --background-activated: #{globals.$ion-bg-neutral-subtlest-press};
+  --background-activated: #{globals.$ion-bg-primary-subtle-press};
   --background-focused: transparent;
-  --background-hover: #{globals.$ion-bg-neutral-subtlest-press};
+  --background-hover: #{globals.$ion-bg-primary-subtle-press};
   --background-hover-opacity: 1;
   --color: #{globals.ion-color(primary, foreground)};
   --ripple-opacity: var(--background-activated-opacity, 1);
   --ripple-color: var(--background-activated);
 }
 
-// Ripple Effect
-// -------------------------------------------------------------------------------
+:host(.button-clear.ion-color) {
+  --background-activated: #{globals.current-color(shade, $subtle: true)};
+  --background-hover: #{globals.current-color(shade, $subtle: true)};
 
-:host(.button-solid.ion-color) ion-ripple-effect {
-  color: globals.current-color(shade);
-}
-
-:host(.button-outline.ion-color) ion-ripple-effect,
-:host(.button-clear.ion-color) ion-ripple-effect {
-  color: globals.$ion-primitives-neutral-200;
+  .button-native {
+    --ion-color-base: #{globals.current-color(foreground)};
+  }
```

Comment:
Is this supposed to be for border color and text color? If so then use the appropriate variables like in a [previous comment](https://github.com/ionic-team/ionic-framework/pull/30764/files#r2492406478).